website/lib/Waypointy.class.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+class Waypointy
+{
+    // database session opened with DBPConnect();
+    public $dblink;
+    // report current activity to stdout
+    public $verbose;
+
+    //~ waypoint attributes
+    public $waypoint;
+    public $lat;
+    public $lon;
+    public $name;
+    public $owner;
+    public $typ;
+    public $cache_type;
+    public $cache_link;
+    public $kraj;
+    public $country;
+
+    public function __construct($dblink, $verbose)
+    {
+        $this->dblink = $dblink;
+        $this->verbose = $verbose;
+    }
+
+    public function getByWaypoint($waypoint)
+    {
+        if (!isset($waypoint) || $waypoint == '') {
+            throw new Exception('Waypoint expected');
+        }
+        $sqlquery = 'SELECT wp.`waypoint`, wp.`lat`, wp.`lon`, wp.`name`, wp.`owner`, wp.`typ`, wpt.`cache_type`, wp.`link`, wp.`kraj`, wpc.`country`'
+               .' FROM `gk-waypointy` wp'
+               .' JOIN `gk-waypointy-country` wpc ON wp.`kraj` = wpc.`kraj`'
+               .' JOIN `gk-waypointy-type` wpt ON wp.`typ` = wpt.`typ`'
+               .' WHERE wp.`waypoint`=\'%s\''
+               .' LIMIT 1';
+        $query = sprintf($sqlquery, mysqli_real_escape_string($this->dblink, $waypoint));
+        $result = mysqli_query($this->dblink, $query);
+        if (!$result) {
+            return false;
+        }
+        $row = mysqli_fetch_row($result);
+        mysqli_free_result($result);
+        list($this->waypoint, $this->lat, $this->lon, $this->name, $this->owner, $this->typ, $this->cache_type, $this->cache_link, $this->kraj, $this->country) = $row;
+
+        return true;
+    }
+
+    public function getByName($waypointName)
+    {
+        if (!isset($waypointName) || $waypointName == '') {
+            throw new Exception('Waypoint name expected');
+        }
+        $sqlquery = 'SELECT wp.`waypoint`, wp.`lat`, wp.`lon`, wp.`name`, wp.`owner`, wp.`typ`, wpt.`cache_type`, wp.`link`, wp.`kraj`, wpc.`country`'
+            .' FROM `gk-waypointy` wp'
+            .' JOIN `gk-waypointy-country` wpc ON wp.`kraj` = wpc.`kraj`'
+            .' JOIN `gk-waypointy-type` wpt ON wp.`typ` = wpt.`typ`'
+            .' WHERE wp.`name` LIKE \'%%%s%%\''
```

Comment:
And here, [use for example](https://stackoverflow.com/a/36593020/944936) `CONCAT('%',?,'%')` (with prepared statement too)